### PR TITLE
Build with disabled features

### PR DESCRIPTION
**Pull request by peterbarker**
This script disables features found in APM_config.h and ensures the build still works.

### CODE REVIEW

**Comment by rmackay9** [comment]
This looks fine to me but I think @OXINARF is probably best placed to review this one.

**Comment by amilcarlucas** [comment]
One of my PRs also adds AC_AVOID_ENABLED to APM_Config.h

**Comment by rmackay9** [comment]
i've merged the PR from @amilcarlucas so the above commit, "Copter: add AC_AVOID_ENABLED to APM_Config.h" can be ignore.

File: Tools/autotest/build-with-disabled-features.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+'''
+Build ArduPilot with various build-time options enabled or disabled
+'''
```
Comment (amilcarlucas) [inline Tools/autotest/build-with-disabled-features.py]: Can you please add usage documentation ?

**Comment by peterbarker** [comment]
On Wed, 21 Feb 2018, Amilcar Lucas wrote:

> +'''
> +Build ArduPilot with various build-time options enabled or disabled
> +'''
> 
> Can you please add usage documentation ?

Done.  We're currently failing to compile when logging is disabled - I 
haven't chased that further.

This is the second time I wrote this script, BTW - I seem to have lost the 
first effort, which was probably better :-)

**Pull request merged**